dist/css-layout.h

```diff
@@ -1266,14 +1277,17 @@ static void layoutNodeImpl(css_node_t* node, float availableWidth, float availab
               remainingFreeSpace / totalFlexGrowFactors * flexGrowFactor);
           }
         }
-        
+
         deltaFreeSpace -= updatedMainSize - childFlexBasis;
-        
+
         if (isMainAxisRow) {
           childWidth = updatedMainSize + getMarginAxis(currentRelativeChild, CSS_FLEX_DIRECTION_ROW);
           childWidthMeasureMode = CSS_MEASURE_MODE_EXACTLY;
-          
-          if (!isStyleDimDefined(currentRelativeChild, CSS_FLEX_DIRECTION_COLUMN)) {
+
+          if (!isUndefined(availableInnerCrossDim) && !isStyleDimDefined(currentRelativeChild, CSS_FLEX_DIRECTION_COLUMN) && heightMeasureMode == CSS_MEASURE_MODE_EXACTLY && getAlignItem(node, currentRelativeChild) == CSS_ALIGN_STRETCH) {
```

Comment:
Maybe add a comment to indicate that this branch and the one below are for an optimization rather than a functional requirement.